Jenkinsfile

```
@@ -329,6 +329,8 @@ pipeline {
                         checkPatch user: GITHUB_USER_USR,
                                    password: GITHUB_USER_PSW,
                                    ignored_files: "src/control/vendor/*:src/include/daos/*.pb-c.h:src/common/*.pb-c.[ch]:src/mgmt/*.pb-c.[ch]:src/iosrv/*.pb-c.[ch]:src/security/*.pb-c.[ch]:*.crt:*.pem:*_test.go:src/cart/_structures_from_macros_.h"
+                        sh label: "codespell",
+			   script: "codespell --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread --builtin clear --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden `git ls-tree --full-tree --name-only HEAD`"
```

(style)  'nd' may be misspelled - perhaps 'and'?
(style)  'ths' may be misspelled - perhaps 'the'?
(style)  'ba' may be misspelled - perhaps 'by'?
(style)  'creat' may be misspelled - perhaps 'create'?
(style)  'te' may be misspelled - perhaps 'the'?
(style)  'mapp' may be misspelled - perhaps 'map'?
(style)  'dout' may be misspelled - perhaps 'doubt'?
(style)  'tre' may be misspelled - perhaps 'tree'?
(style)  'reord' may be misspelled - perhaps 'record'?
(style)  'mimick' may be misspelled - perhaps 'mimic'?
(style)  'cloneable' may be misspelled - perhaps 'clonable'?
(style)  'keypair' may be misspelled - perhaps 'key pair'?
(style)  'bject' may be misspelled - perhaps 'object'?

---

How do we get codespell not to check the spelling on the exception list?  It seems like every patch that updates the whitelist is itself going to fail the codespell.

Can the whitelist be put into a file, perhaps `cat`ted into the codespell commandline option and the whiltelist file excepted from codespell checking?

```suggestion
			   script: "codespell --ignore-words-list $(cat codespell-whitelist) --builtin clear --skip codespell-whitelist,*.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden `git ls-tree --full-tree --name-only HEAD`"
```

Ultimately we **really** need to reduce the number of exceptions we are needing to allow through CI.  As you have seen, we are becoming complacent with allowing things to fail and land anyway and because of it we are introducing new failures that shouldn't have been allowed.

---

Good questions.  Having the check here will check the entire tree but has really bad reporting, so what I think we should do is get this PR to a point where we're happy with the corrections, remove this change and then land it.  I was hopeful that once we had checking of PRs (which we've had for some weeks now) that we would no longer be introducing regressions, and that would be the end of it.

---

I'm not sure I'm quite grasping

> remove this change 

means.  What would we be removing?